fetch.bs

```diff
@@ -3237,7 +3250,103 @@ through TLS using ALPN. The protocol cannot be spoofed through HTTP requests in
+to attempt to fetch them over <a>potentially trustworthy URL</a>s. If an upgraded request
+fails with a network error, it is retried over the original URL.
+
+<p>The HTTPS upgrading algorithm consists of Upgrade and Fallback algorithms, with the following
+steps:
+
+<h4 id=https-upgrades-upgrade>HTTPS upgrade algorithm</h4>
 
+To <dfn>upgrade an HTTP request</dfn> given a <a for=/>request</a> <var>request</var>, run these
+steps:
+
+<ol>
+  <li>
+    <p>If one or more of the following conditions are met, return:
+    <ul>
+      <li><p><var>request</var>'s <a for="request">destination</a> is not "<code>document</code>"
+
+      <li><p><var>request</var>'s <a for="request">method</a> is not "<code>GET</code>"
+
+      <li><p><var>request</var>'s <a for="request">URL</a>'s <a for="url">scheme</a> is not
+      "<code>http</code>"
+
+      <li><p><var>request</var>'s <a for="request">URL</a>'s <a for="url">host</a> is exempted from
+      upgrades in an <a>implementation-defined</a> way.
+
+      <p class="example">If <a for=url>host</a> is a non-registrable or non-assignable domain name
+      such as .local or an IP address that falls in a range reserved for non-publicly routable
+      networks, the implementation might return without modifying <var>request</var>.
+    </ul>
+  </li>
+
+  <li><p>If <var>request</var>'s <code>https-upgrade-fallback-url</code> is non-null, clear the
+  <code>https-upgrade</code> and <code>https-upgrade-fallback-url</code> flags and return.
+
+  <p class="note">This is a fallback request that shouldn't be upgraded again.
+
+  <li><p>Otherwise, upgrade the request by setting the following fields:
+    <ul>
+      <li>Set <code>https-upgrade-fallback-url</code> to <var>request</var>'s <a for="request">URL</a>.
+      <li>Set <var>request</var>'s <a for="request">URL</a>'s <a for="url">scheme</a> to "<code>https</code>".
+      <li>Set <code>https-upgrade</code> flag to true.
+    </ul>
+  </li>
+</ol>
+
+<h4 id=https-upgrades-fallback>Fallback algorithm</h4>
+
+To run <dfn>HTTPS upgrade fallback</dfn> given a <a for=/>request</a>
+<var>request</var> and <a for=/>response</a> <var>response</var>, run these steps:
+
+<ol>
+  <li><p>If <var>request</var>'s <code>https-upgrade</code> flag is not set, return
+  <var>response</var>.
+
+  <p class="note">This means <var>request</var> was not upgraded.
+
+  <li>
+    <p class="note">The request was upgraded.
+
+    <p>If <var>response</var> is a network error, run the following steps:
+    <p class="note">This means that the upgrade failed and initiates a fallback load.
+     <ol>
+      <li><p>Let <var>fallbackResponse</var> be a new <a>Response</a> with its <code>Location</code> header set to
+      <code>https-upgrade-fallback-url</code>, and its <a for="response">status</a> set to 307.
+      <li>Return <var>fallbackResponse</var>.
+     </ol>
+
+  <li><p>Otherwise, the upgrade was successful. Return <var>response</var>.
```

Comment:
No need for "otherwise" here as everything above early returns. "upgrade was successful" should be a note.

Reply:
Done